constraint-evaluation-generator/src/substitution.rs

```diff
@@ -241,93 +219,144 @@ impl Substitutions {
     }
 
     fn base_single_row_substitutions(
-        indices: &[usize],
+        section_start_index: usize,
         substitutions: &[TokenStream],
     ) -> TokenStream {
-        assert_eq!(indices.len(), substitutions.len());
+        let num_substitutions = substitutions.len();
+        let indices = (0..num_substitutions).collect_vec();
         if indices.is_empty() {
             return quote!();
         }
         quote!(
-            master_base_table.rows_mut().into_iter().for_each(|mut row| {
-            #(
-            let (base_row, mut det_col) =
-                row.multi_slice_mut((s![..#indices],s![#indices..=#indices]));
-            det_col[0] = #substitutions;
-            )*
-            });
+            let (original_part, mut current_section) =
+                master_base_table.multi_slice_mut(
+                    (
+                        s![.., 0..#section_start_index],
+                        s![.., #section_start_index..#section_start_index+#num_substitutions],
+                    )
+                );
+            Zip::from(original_part.rows())
+                .and(current_section.rows_mut())
+                .par_for_each(|original_row, mut section_row| {
+                    let mut base_row = original_row.to_owned();
+                    #(
+                        let det_col = section_row.get_mut(#indices).unwrap();
+                        *det_col = #substitutions;
```

Comment:
I can't get that to work. The test fails. (Can't verify proof.) Compiling the auto-generated code also generates this warning: 
```
warning: value assigned to `det_col` is never read
```

Reply:
[This seems to work](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=5437498a908653bf8ec244029d874687). Was that how you tried it?

Reply:
That is not how I tried it. Thanks for the working example! Fixed in aff29a3.

We could assign `#substitutions` to `section_row[#indices]` directly without passing via `det_col` but we still need to populate the current row by appending a single field element, and coercing `section_row[#indices]` into the right type seems suboptimal because difficult to read.

Reply:
Hang on, let me use `push` instead of `append`; maybe that gives something cleaner still ...